python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -791,11 +792,19 @@ def resolve_partitionStats(self, graphene_info) -> Optional[GraphenePartitionSta
             if materialized_partition_subset is None:
                 check.failed("Expected partitions subset for a partitioned asset")
 
+            failed_partition_subset = get_failed_partitions_subset(
+                graphene_info.context.instance,
+                asset_key,
+                self._dynamic_partitions_loader,
+                partitions_def_data.get_partitions_definition(),
+            )
+
             return GraphenePartitionStats(
                 numMaterialized=len(materialized_partition_subset),
                 numPartitions=partitions_def_data.get_partitions_definition().get_num_partitions(
                     dynamic_partitions_store=self._dynamic_partitions_loader
                 ),
+                numFailed=len(failed_partition_subset),
```

Comment:
Thoughts on not counting a partition as materialized if it was failed more recently than materialized? I think it might be confusing if a partition can be counted both as materialized and failed. I think if it failed, it should not be counted as "materialized".

Reply:
On `GrapheneAssetNode` theres also a `materializedPartitions` resolver that resolves the partition keys of the materialized partitions. I think we'll want that resolver to encompass failed partitions, though that can happen in a future PR. It might be easier to group those changes together with the `PartitionStats` changes in a separate PR.